nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/PutAzureBlobStorage_v12.java

```diff
@@ -200,7 +199,13 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
 
             long transferMillis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNanos);
             String transitUri = attributes.get(ATTR_NAME_PRIMARY_URI);
-            session.getProvenanceReporter().send(flowFile, transitUri, transferMillis);
+            if (fileResourceFound.isPresent()) {
+                final FileResource fileResource = fileResourceFound.get();
+                session.getProvenanceReporter().fetch(flowFile, transitUri, transferMillis);
```

Comment:
The `fetch()` method is intended to indicate that the content of the FlowFile was replaced with an external source, but that is not the case in this scenario. The output FlowFile remains unchanged, so `FETCH` does not seem to apply. The `receive()` method seems more appropriate as indicates the introduction of data, as the content of the FlowFile remains unchanged.